### PR TITLE
Lighthouse: Render a Box instead of a Link when no `href` prop exists

### PR DESCRIPTION
**Pull request by ianrohde**
## Issue
This was flagged on a Lighthouse run for HTML semantics. We are rendering NavTabs without href attributes.

## CR/QA
This change renders a `Box` component instead of a `Link` when the href URL doesn't exist.

<details>
<summary>After Screenshot</summary>

Screenshot 2023-07-20 at 2 03 02 PM

</details>

Closes https://github.com/iFixit/react-commerce/issues/1840

### CODE REVIEW

**Comment by jordycosta** [comment]
Since this removes the `Link` component if no href exists, I think we could drop the _isDisabled_ prop

**Comment by ianrohde** [comment]
isDisabled provides specific styling. There was a conditional that rendered it on the Link, so with that changed, we need to apply it on the Box.

**Comment by deltuh-vee** [comment]
QA 🎬 
When Parts is disabled, we're no longer getting penalized on Lighthouse runs for having an uncrawlable link.

**Comment by deltuh-vee** [comment]
dev_block ✊ 
Parts is appearing twice.
image

File: frontend/templates/troubleshooting/index.tsx [hunk 1]
```diff
@@ -504,20 +504,26 @@ function NavTabs({
...
+            </Link>
+         ) : (
+            <Box className="isDisabled" {...notSelectedStyleProps}>
+               Parts
+            </Box>
+         )}
+
+         {deviceGuideUrl ? (
+            <Link {...notSelectedStyleProps} href={deviceGuideUrl}>
+               Parts
```
Comment (jordycosta) [inline frontend/templates/troubleshooting/index.tsx:519]:
```suggestion
               Guides
```

**Comment by ianrohde** [comment]
Thanks for catching that!

File: frontend/templates/troubleshooting/index.tsx [hunk 2]
```diff
@@ -504,20 +504,26 @@ function NavTabs({
...
+         {deviceGuideUrl ? (
+            <Link {...notSelectedStyleProps} href={deviceGuideUrl}>
+               Guides
+            </Link>
+         ) : (
+            <Box className="isDisabled" {...notSelectedStyleProps}>
+               Guides
+            </Box>
+         )}
+
```
Comment (aburke07) [inline frontend/templates/troubleshooting/index.tsx:526]: Completely optional, but we can get rid of the duplicate ternary if we wrap the two links in a react fragment and the two boxes in a react fragment.

**Review by aburke07 — APPROVED** [review]
CR :alien: with an optional comment

File: frontend/templates/troubleshooting/index.tsx (hunk 2 above)
Comment (ianrohde) [reply frontend/templates/troubleshooting/index.tsx:526]: The conditionals target `devicePartsUrl` and `deviceGuideUrl`, so unfortunately it's not that simple. I was going to get fancy and break this out into a function, but in this instance, I just didn't think it was worth it.

**Comment by deltuh-vee** [comment]
QA 🎬 
A Box component is rendered instead of a Link when the href URL doesn't exist for Parts.

**Pull request merged**